content/sensu-go/5.0/reference/rbac.md

```diff
@@ -1,132 +1,298 @@
-such as `development`, `staging` and `production`. An environment belongs to a
-single organization. Every organization has a `default` environment.
+To update the description for a namespace:
+
+{{< highlight shell >}}
+sensuctl namespace update [NAMESPACE-NAME]
+{{</highlight>}}
+
+To delete a namespace:
+
+{{< highlight shell >}}
+sensuctl namespace delete [NAMESPACE-NAME]
+{{</highlight>}}
+
+_WARNING: This deletes every resource definition associated with the namespace._
+
+To get help managing namespaces with sensuctl:
+
+{{< highlight shell >}}
+sensuctl namespace help
+{{</highlight>}}
+
+### Namespace specification
 
 #### Attributes
 
 name         | 
 -------------|------ 
-description  | The name of the environment. Cannot contain special characters.
+description  | The name of the namespace. Names can contain alphanumeric characters and hyphens, but must begin and end with an alphanumeric character.
 required     | true 
 type         | String
-example      | {{< highlight shell >}}"name": "default"{{</highlight>}}
+example      | {{< highlight shell >}}"name": "ops"{{</highlight>}}
 
 description  | 
 -------------|------ 
-description  | Description of the environment 
+description  | Description of the namespace
 required     | false 
 type         | String
-example      | {{< highlight shell >}}"description": "Default environment"{{</highlight>}}
+example      | {{< highlight shell >}}"description": "Ops team production monitoring"{{</highlight>}}
 
-organization | 
--------------|------ 
-description  | The name of the organization the environment belongs to. 
-required     | false 
-type         | String
-default      | current environment value configured for `sensuctl` (ie `default`)
-example      | {{< highlight shell >}}"organization": "default"{{</highlight>}}
+## Resources
+Permissions within Sensu are scoped to resource types, like checks, handlers, and users.
+All resource types (with the exceptions of namespaces, users, and roles) must belong to a single namespace.
 
-### User
+### Resource types
 
-A user represents a person or an agent which interacts with Sensu.
+| Type | Description |
+|---|---|
+| `*` | All resources within Sensu. **The `*` type takes precedence over other rules within the same role.** If you wish to deny a certain type, you can't use the `*` type and must explicitly allow every type required |
+| `assets` | [Asset][5] resources within a namespace |
+| `checks` | [Check][6] resources within a namespace |
+| `entities` | [Entity][7] resources within a namespace |
+| `events` | [Event][8] resources within a namespace |
+| `handlers` | [Handler][9] resources within a namespace |
+| `hooks` | [Hook][10] resources within a namespace |
+| `mutators` | [Mutator][11] resources within a namespace |
+| `namespace` | [Namespaces][12] |
+| `roles` | [Roles][13] |
+| `silenced` | [Silencing][14] resources within a namespace |
+| `users` | [Users][15] |
```

Comment:
Should there be a filters type included here?

Reply:
The graphql stuff might not be the most up-to-date reference. We have a "registry" [here](https://github.com/sensu/sensu-go/blob/master/types/typemap.go#L11) but the challenge is that some of these types are embedded within others (e.g. a _Rule_ is part of a _Role_) and others are meant for internal use (e.g. _Claims_). The [APId routers](https://github.com/sensu/sensu-go/tree/master/backend/apid/routers) are probably a good place to get started too.

Also, we introduced the notion of a `kind`, so we have to make a distinction between a `type`, a `kind` and a `resource`. Based on my understanding of Kubernetes, here's my definition for each of these terms but it's confusing and should be improved:

| Term | Description |
|---|---|
| object | Objects are persistent entities* in the Sensu system ([source](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#understanding-kubernetes-objects)) |
| resource | A representation of a system entity*, sent or retrieved as JSON via HTTP to the server. Users will specify _resources_ in their RBAC rules (e.g. entities, checks) | 
| type | The literal Go type. The term "type" should be used for distinguishing sub-categories within objects (e.g. An event object has a `check` and an `entity` type). | 
| kind | The name of a particular object schema (e.g. Check or Hooks kinds that have different attributes and properties). Users will need to specify a _kind_ when creating resources via `sensuctl create` for example. | 

*The term `entity` is really confusing here since we have Sensu entities, so we should probably find a better term here.

Anyway, I think we are missing the following resources:

- `cluster` (to manage the cluster)
- `filters`
- `extensions`
- `clusterroles`
- `clusterrolebindings`
- `rolebindings`

Reply:
Updated!